### ISSUE TITLE
Video progress resets when switching player - Workaround #7668 doesn't work for playlists

### ISSUE THREAD

**Issue by Dangelman** [TeamNewPipe/NewPipe#7990]
### Checklist

- [X] I am able to reproduce the bug with the [latest version](https://github.com/TeamNewPipe/NewPipe/releases/latest).
- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Affected version

0.22.1

### Steps to reproduce the bug

1. Press "Play all" in a playlist or enqueue multiple videos
2. After watching a video for a couple of minutes, switch from main to pop-up
(3. Switch back from pop-up to main)

### Expected behavior

Video progress is saved, like with single videos (thanks to workaround #7668).

### Actual behavior

Enqueued videos have their progress reset when switching from main to pop-up.
Additionally, switching back to main from pop-up will create an error

## Exception
* __User Action:__ play stream
* __Request:__ Player error[type=1] occurred while playing https://www.youtube.com/watch?v=dDakzFJfZ4c
* __Content Country:__ US
* __Content Language:__ en
* __App Language:__ en
* __Service:__ YouTube
* __Version:__ 0.22.1
* __OS:__ Linux Android 11 - 30
<details><summary><b>Crash log </b></summary><p>

```
com.google.android.exoplayer2.ExoPlaybackException: MediaCodecVideoRenderer error, index=0, format=Format(1, null, null, video/avc, avc1.640020, -1, null, [1080, 1080, -1.0], [-1, -1]), format_supported=YES
	at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:555)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:223)
	at android.os.HandlerThread.run(HandlerThread.java:67)
Caused by: com.google.android.exoplayer2.mediacodec.MediaCodecRenderer$DecoderInitializationException: Decoder init failed: OMX.qcom.video.decoder.avc, Format(1, null, null, video/avc, avc1.640020, -1, null, [1080, 1080, -1.0], [-1, -1])
	at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.maybeInitCodecWithFallback(MediaCodecRenderer.java:1051)
	at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.maybeInitCodecOrBypass(MediaCodecRenderer.java:606)
	at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.onInputFormatChanged(MediaCodecRenderer.java:1465)
	at com.google.android.exoplayer2.video.MediaCodecVideoRenderer.onInputFormatChanged(MediaCodecVideoRenderer.java:713)
	at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.readSourceOmittingSampleData(MediaCodecRenderer.java:998)
	at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.render(MediaCodecRenderer.java:830)
	at com.google.android.exoplayer2.ExoPlayerImplInternal.doSomeWork(ExoPlayerImplInternal.java:945)
	at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:478)
	... 3 more
Caused by: java.lang.IllegalArgumentException: The surface has been released
	at android.media.MediaCodec.native_configure(Native Method)
	at android.media.MediaCodec.configure(MediaCodec.java:2127)
	at android.media.MediaCodec.configure(MediaCodec.java:2043)
	at com.google.android.exoplayer2.mediacodec.SynchronousMediaCodecAdapter$Factory.createAdapter(SynchronousMediaCodecAdapter.java:50)
	at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.initCodec(MediaCodecRenderer.java:1137)
	at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.maybeInitCodecWithFallback(MediaCodecRenderer.java:1044)
	... 10 more

```
</details>
<hr>

**Comment by Stypox** [comment]
I can confirm 🙈

**Comment by tsiflimagas** [comment]
This seems to have been fixed by #8020.
